opentelemetry-otlp/src/exporter/tonic/mod.rs

```diff
@@ -207,20 +220,107 @@ impl TonicExporterBuilder {
         };
 
         #[cfg(feature = "tls")]
-        let channel = match self.tonic_config.tls_config {
-            Some(tls_config) => endpoint
-                .tls_config(tls_config)
-                .map_err(crate::Error::from)?,
-            None => endpoint,
+        {
+            let insecure = config.insecure.unwrap_or_else(|| {
```

Comment:
Please correct me - there would always be a default value associated with the `insecure`, which means else part will never execute?